retrofit-converters/protobuf-nano/src/main/java/retrofit2/converter/protobuf/nano/ProtoResponseBodyConverter.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2.converter.protobuf.nano;
+
+import com.google.protobuf.nano.InvalidProtocolBufferNanoException;
+import com.google.protobuf.nano.MessageNano;
+import java.io.IOException;
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+
+/**
+ * * @author Keith Banks - Architect at siilo.com
+ */
+final class ProtoResponseBodyConverter<T extends MessageNano>
+        implements Converter<ResponseBody, T> {
+  private final T instance;
+  ProtoResponseBodyConverter(T instance) {
+    this.instance = instance;
+  }
+
+  @Override public T convert(ResponseBody value) throws IOException {
+    try {
+      return MessageNano.mergeFrom(instance, value.bytes());
```

Comment:
Does this update `instance` or return a new copy? Because if it's the latter then a new instance needs created for every invocation of this method. We only create the converters once but call them many times.